llm.el

```diff
@@ -128,6 +131,52 @@ REQUIRED is whether this is required or not."
   type
   required)
 
+(cl-defstruct llm-media
+  "Contains media that can be sent as part of an interaction.
+
+MIME-TYPE is a string containing the mime type of the media.
+
+DATA is a (binary) string containing the data."
```

Comment:
Can you add details on the encoding of this string?  Should it be a multibyte string?  Unibyte?

Reply:
I hadn't thought about this because the encoding is, I think, just ignored by base64-encode-string, but it seems more sensible to use unibyte for binary data, so I settled on that and adjusted the code as appropriate to ensure this.